pkg/nfd-master/nfd-master.go

```diff
@@ -265,19 +291,30 @@ func (s *labelerServer) SetLabels(c context.Context, r *pb.SetLabelsRequest) (*p
 	}
 	stdoutLogger.Printf("REQUEST Node: %s NFD-version: %s Labels: %s", r.NodeName, r.NfdVersion, r.Labels)
 
-	labels := filterFeatureLabels(r.Labels, s.args.ExtraLabelNs, s.args.LabelWhiteList)
+	labels, extendedResources := filterFeatureLabels(r.Labels, s.args.ExtraLabelNs, s.args.LabelWhiteList, s.args.ResourceLabels)
 
 	if !s.args.NoPublish {
-		// Advertise NFD worker version and label names as annotations
-		keys := make([]string, 0, len(labels))
-		for k, _ := range labels {
-			keys = append(keys, k)
+		// Advertise NFD worker version, label names and extended resources as annotations
+		labelKeys := make([]string, 0, len(labels))
+		for k := range labels {
+			labelKeys = append(labelKeys, k)
 		}
-		sort.Strings(keys)
+		sort.Strings(labelKeys)
+
+		extendedResourceKeys := []string{}
+		for key, value := range extendedResources {
+			if value != "" {
```

Comment:
With the change I proposed in `filterFeatureLabels` you could remove this check, I think